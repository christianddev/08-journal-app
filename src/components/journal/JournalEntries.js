import React from 'react';
import JournalEntry from './JournalEntry';

const JournalEntries = () => {
  const entries = [1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16];
  return (
    <div className="journal__entries">
      {
        entries.map((e) => <JournalEntry key={e} />)
      }
    </div>
  )
}

export default JournalEntries;
