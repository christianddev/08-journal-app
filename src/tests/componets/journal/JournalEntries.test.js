import React from 'react';
import { mount } from "enzyme"
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom'
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import '@testing-library/jest-dom';

import JournalEntries from '../../../components/journal/JournalEntries';

const middlewares = [thunk]
const mockStore = configureStore(middlewares)
const initState = {
  auth:{},
  ui:{
    loading: false,
    msgError: null
  },
  notes: {
    active: {
      id: 'abc',
      title: 'titleXD',
      body: 'bodyXD'
    },
    notes:  [
      {
        id: 'abc',
        title: 'titleXD',
        body: 'bodyXD'
      }
    ]
  }
}
let store = mockStore(initState)
store.dispatch = jest.fn()

const wrapper = mount (
  <Provider store={store}>
    <MemoryRouter>
      <JournalEntries />
    </MemoryRouter>
  </Provider>
)

describe('JournalEntries.js', () => {
  test('should show', () => {
    expect(wrapper).toMatchSnapshot()
  })

})
