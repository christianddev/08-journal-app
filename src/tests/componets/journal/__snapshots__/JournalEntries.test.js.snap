// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JournalEntries.js should show 1`] = `
<div
  className="journal__entries"
>
  <div
    className="journal__entry animate__animated animate__bounce animate__fadeIn animate__faster"
    onClick={[Function]}
  >
    <div
      className="journal__entry-body"
    >
      <p
        className="journal__entry-title"
      >
        titleXD
      </p>
      <p
        className="journal__entry-content"
      >
        bodyXD
      </p>
    </div>
    <div
      className="journal__entry-date-box"
    >
      <span>
        Saturday
      </span>
      <h4>
        23rd
      </h4>
    </div>
  </div>
</div>
`;
